roxie/udplib/udpsha.hpp

```diff
@@ -202,14 +202,29 @@ class simple_queue
 
 class flowType {
 public:
-    enum flowCmd : unsigned short { ok_to_send, request_received, request_to_send, send_completed, request_to_send_more, max_flow_cmd };
+    enum flowCmd : unsigned short
+    {
+    //The first items in the flow command are also used as states in the receiver code:
+        send_completed,         // no activity from the sender - wating for a request to send
+        request_to_send,        // permit to be send has been requested but not granted
+        ok_to_send,             // permit has been granted
+
+    // The following messages
+        request_received,       // acknowledge request to send from the sender
+        send_start,             // about to send data - indicate how many packets are actually going to be send
+        request_to_send_more,   // equivalent to send_completed followed by a request_to_send
```

Comment:
May want to uncombine these if one should go on data socket (so as not to overtake associated data) but other should not.